fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/api/event/registry/FabricRegistryBuilder.java

```diff
@@ -50,31 +55,33 @@
 	 * @param <R> The registry type
 	 * @return An instance of FabricRegistryBuilder
 	 */
-	public static <T, R extends Registry<T>> FabricRegistryBuilder<T, R> from(R registry) {
+	public static <T, R extends MutableRegistry<T>> FabricRegistryBuilder<T, R> from(R registry) {
 		return new FabricRegistryBuilder<>(registry);
 	}
 
 	/**
 	 * Create a new {@link FabricRegistryBuilder} using a {@link SimpleRegistry}, the registry has the {@link RegistryAttribute#MODDED} attribute by default.
 	 *
-	 * @param type A class matching the type being stored in the registry
+	 * @param registryKey The RegistryKey to create the registry
+	 * @param lifecycle The {@link Lifecycle} used to create the registry
 	 * @param <T> The type stored in the Registry
 	 * @return An instance of FabricRegistryBuilder
 	 */
-	public static <T> FabricRegistryBuilder<T, SimpleRegistry<T>> createSimple(Class<T> type) {
-		return from(new SimpleRegistry<>());
+	public static <T> FabricRegistryBuilder<T, SimpleRegistry<T>> createSimple(class_5321<Registry<T>> registryKey, Lifecycle lifecycle) {
+		return from(new SimpleRegistry<>(registryKey, lifecycle));
 	}
 
 	/**
 	 * Create a new {@link FabricRegistryBuilder} using a {@link DefaultedRegistry}, the registry has the {@link RegistryAttribute#MODDED} attribute by default.
 	 *
-	 * @param type A class matching the type being stored in the registry
+	 * @param registryKey The RegistryKey to create the registry
+	 * @param lifecycle The {@link Lifecycle} used to create the registry
 	 * @param defaultId The default registry id
 	 * @param <T> The type stored in the Registry
 	 * @return An instance of FabricRegistryBuilder
 	 */
-	public static <T> FabricRegistryBuilder<T, DefaultedRegistry<T>> createDefaulted(Class<T> type, Identifier defaultId) {
-		return from(new DefaultedRegistry<>(defaultId.toString()));
+	public static <T> FabricRegistryBuilder<T, DefaultedRegistry<T>> createDefaulted(Identifier defaultId, class_5321<Registry<T>> registryKey, Lifecycle lifecycle) {
```

Comment:
Imo we should use an identifier. You cannot prevent people from passing botched registry keys with registry reference other than `minecraft:root`, while an identifier only specifying the value of the registry key is always safe.